front/src/app/services/user/user.service.ts

```diff
@@ -17,16 +19,40 @@ export class UserService {
 
   constructor(private http: HttpClient) {
     this._userDataLoaded = null;
+    this._userData = {
+      id: -1,
+      username: '',
+      email: '',
+      color_config: DEFAULT_COLORS,
+    };
   }
 
   private async loadUserData(): Promise<void> {
     try {
-      this._userData = await this.http.get<User>(`${API_USER}/user_data/`).toPromise();
+      const userData = await this.http.get<User>(`${API_USER}/user_data/`).toPromise();
+      if (!userData) {
+        throw new Error('No user data found');
+      }
+      this._userData = {
+        id: Object.freeze(userData.id),
+        email: Object.freeze(userData.email),
+        username: Object.freeze(userData.username),
+        color_config: userData.color_config,
```

Comment:
Freeze the color_config array since changing the color config during the game is unnecessary

Reply:
no because we need to be able to change it